charts/super-agent-deployment/templates/_helpers.tpl

```diff
@@ -52,3 +52,32 @@ readOnlyRootFilesystem: true
     {{- toYaml $defaults -}}
 {{- end -}}
 {{- end -}}
+
+{{- /*
+Check if authSecret.create is explicitly set to true. If authSecret is not empty and create is not defined, default it to false.
+*/ -}}
+{{- define "newrelic-super-agent.authSecretDefaults" -}}
```

Comment:
the name authSecretDefaults is not that clear 😕

Reply:
Updated to `shouldCreateAuthSecret`.